src/WebJobs.Script.WebHost/Extensions/ScriptHostManagerExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Extensions;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public static class ScriptHostManagerExtensions
+    {
+        public static async Task WaitForRunningHostAsync(this IScriptHostManager hostManager, ScriptApplicationHostOptions applicationHostOptions, HttpContext httpContext, int timeoutSeconds = ScriptConstants.HostTimeoutSeconds, int pollingIntervalMilliseconds = ScriptConstants.HostPollingIntervalMilliseconds, ActionExecutionDelegate next = null)
```

Comment:
I factored this "wait for running host" logic out of the RequiresRunningHost attribute so it could be reused. To do so, I had to move the ScriptHostManagerExtensions helper to WebHost level since it operates on WebHost only types.